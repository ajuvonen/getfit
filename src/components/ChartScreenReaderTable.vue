<script lang="ts" setup>
defineProps<{
  id: string,
  title: string,
  labels: string[],
  data: number[],
}>();
</script>
<template>
  <table class="d-sr-only" :id="id">
    <caption>{{ title }}</caption>
    <thead>
      <tr>
        <th v-for="label in labels" :key="label">{{ label }}</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td v-for="count in data" :key="count">{{ count }}</td>
      </tr>
    </tbody>
  </table>
</template>
