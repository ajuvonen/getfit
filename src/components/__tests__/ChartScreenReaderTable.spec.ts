import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import ChartScreenReaderTable from '@/components/ChartScreenReaderTable.vue';

describe('ChartScreenReaderTable', () => {
  it('mounts', () => {
    const wrapper = mount(ChartScreenReaderTable, {
      props: {
        title: 'Amount of trainings',
        id: 'my-chart',
        labels: ['Running', 'Sprint', 'Swimming'],
        data: [1, 2, 3],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('table').classes()).toContain('d-sr-only');
  });
});
