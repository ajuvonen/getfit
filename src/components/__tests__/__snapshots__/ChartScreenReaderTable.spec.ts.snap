// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartScreenReaderTable > mounts 1`] = `
"<table class="d-sr-only" id="my-chart">
  <caption>Amount of trainings</caption>
  <thead>
    <tr>
      <!--v-if-->
      <th>Running</th>
      <th>Sprint</th>
      <th>Swimming</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>2</td>
      <td>3</td>
    </tr>
  </tbody>
</table>"
`;

exports[`ChartScreenReaderTable > mounts with row headers 1`] = `
"<table class="d-sr-only" id="my-chart">
  <caption>Amount of trainings</caption>
  <thead>
    <tr>
      <td></td>
      <th>Running</th>
      <th>Sprint</th>
      <th>Swimming</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Week 1</th>
      <td>1</td>
      <td>2</td>
      <td>3</td>
    </tr>
    <tr>
      <th>Week 2</th>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>Week 3</th>
      <td>3</td>
      <td>3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>"
`;
